src/detect-dns-query-name.c

```diff
@@ -0,0 +1,182 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * Detect keyword for DNS query names: dns.query.name
+ */
+
+#include "detect.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-prefilter.h"
+#include "detect-engine-content-inspection.h"
+#include "detect-dns-query-name.h"
+#include "util-profiling.h"
+#include "rust.h"
+
+static int DetectSetup(DetectEngineCtx *, Signature *, const char *);
```

Comment:
You could avoir declarations by putting the functions in the right order

Reply:
I thought at one point we were opinionated about prototypes even for static functions.

Reply:
Oh, I did not know : 
- I was not rejected for proposing commits without pre-prototypes
- My 2 cents : I think it is cleaner without these repetitions

Reply:
I can't find any stuff in our guides around this. I also think its cleaner without the repetition so will do that in the next version.

Reply:
yeah prefer reordering to reduce unnecessary prototypes